### PR TITLE
[XR] multi-pointer screen coordinates projection

### PR DESCRIPTION
**Pull request by RaananW**
This PR will allow passing the projected screen coordinates of controllers in XR.

The data will be available in `event.screenX` and `event.screenY` (if available at all).

### CODE REVIEW

File: src/XR/features/WebXRControllerPointerSelection.ts [hunk 1]
```diff
@@ -392,9 +393,16 @@ export class WebXRControllerPointerSelection extends WebXRAbstractFeature {
                 if (camera) {
                     camera.viewport.toGlobalToRef(scene.getEngine().getRenderWidth(), scene.getEngine().getRenderHeight(), this._viewportRef);
                     Vector3.ProjectToRef(controllerGlobalPosition, this._identityMatrix, scene.getTransformMatrix(), this._viewportRef, this._screenCoordinatesRef);
-
-                    scene.pointerX = this._screenCoordinatesRef.x;
-                    scene.pointerY = this._screenCoordinatesRef.y;
+                    // stay safe
```
Comment (sebavan) [inline src/XR/features/WebXRControllerPointerSelection.ts:412]: What else than number could be in ?

File: src/XR/features/WebXRControllerPointerSelection.ts [hunk 2]
```diff
@@ -487,6 +495,10 @@ export class WebXRControllerPointerSelection extends WebXRAbstractFeature {
             if (!controllerData.pick) {
                 return;
             }
+            if (controllerData.screenCoordinates) {
```
Comment (sebavan) [inline src/XR/features/WebXRControllerPointerSelection.ts:514]: could we factorize this block ? it looks similar in the 4 cases ?

**Review by sebavan — APPROVED** [review]
Only a couple Nits

File: src/XR/features/WebXRControllerPointerSelection.ts (hunk 1 above)
Comment (RaananW) [reply src/XR/features/WebXRControllerPointerSelection.ts:412]: NaN :-)

File: src/XR/features/WebXRControllerPointerSelection.ts (hunk 2 above)
Comment (RaananW) [reply src/XR/features/WebXRControllerPointerSelection.ts:514]: The pointerEvent object is reused on a lot of those, but is not reused between different types of ray modes.

Comment (RaananW) [reply src/XR/features/WebXRControllerPointerSelection.ts:514]: let me see if I can find a nice way to unify those.

**Pull request merged**